elements/pfe-navigation/src/pfe-navigation-item.js

```diff
@@ -0,0 +1,429 @@
+  
+      this.focus();
+  
+      // Dispatch the event
+      this.dispatchEvent(
+        new CustomEvent(`${this.tag}:close`, {
+          detail: {},
+          bubbles: true,
+          composed: true
+        })
+      );
+    }
+  
+    toggle(event) {
+      if (event) event.preventDefault();
+  
+      if (this.visible && !this.expanded) {
+        this.open(event);
+      } else {
+        this.close(event);
+      }
+    }
+  
+    constructor() {
+      super(PfeNavigationItem);
+  
+      // States
+      this.nested = false;
+      this.expanded = false;
+  
+      // Objects
+      this.trigger = null;
+      this.tray = null;
+      this.directLink = null;
+      this.linkUrl = null;
+  
+      // Lists
+      this.nestedItems = [];
+  
+      // Shadow elements
+      this._trigger = this.shadowRoot.querySelector(`.${this.tag}__trigger`);
+      this._icon = this.shadowRoot.querySelector("pfe-icon");
+      this._tray = this.shadowRoot.querySelector(`.${this.tag}__tray`);
+  
+      // Externally accessible events
+      this.close = this.close.bind(this);
+      this.open = this.open.bind(this);
+      this.toggle = this.toggle.bind(this);
+  
+      // Initializers
+      this._init__trigger = this._init__trigger.bind(this);
+      this._init__tray = this._init__tray.bind(this);
+  
+      // Event handlers
+      this._keydownHandler = this._keydownHandler.bind(this);
+      this._keyupHandler = this._keyupHandler.bind(this);
+      this._navigateToUrl = this._navigateToUrl.bind(this);
+      this._directLinkHandler = this._directLinkHandler.bind(this);
+    }
+  
+    connectedCallback() {
+      super.connectedCallback();
+  
+      this._init__trigger();
+      this._init__tray();
+  
+      // Add a slotchange listeners to the lightDOM elements
+      if (this.trigger) this.trigger.addEventListener("slotchange", this._init__trigger);
+      if (this.tray) this.tray.addEventListener("slotchange", this._init__tray);
+    }
+  
+    attributeChangedCallback(attr, oldValue, newValue) {
+      super.attributeChangedCallback(attr, oldValue, newValue);
+    }
+  
+    disconnectedCallback() {
+      this.trigger.removeEventListener("slotchange", this._init);
+  
+      if (this.tray) {
+        this.tray.removeEventListener("slotchange", this._init);
+  
+        this.removeEventListener("keydown", this._keydownHandler);
+        this.removeEventListener("keyup", this._exit);
+  
+        this._trigger.removeEventListener("click", this.toggle);
+        this._trigger.removeEventListener("keyup", this._keyupHandler);
+      } else {
+        this._trigger.removeEventListener("click", this._navigateToUrl);
+        this._trigger.removeEventListener("keyup", this._directLinkHandler);
+      }
+    }
+  
+    _init__trigger() {
+      // If no slots have been assigned, assign it to the trigger slot
+      const unassigned = [...this.children].filter(child => !child.hasAttribute("slot"));
```

Comment:
I'm pretty sure this is going to fail in Angular but I haven't tested it yet. `_init__trigger` is being called in the `connectedCallback` and I'm pretty sure that `this.children` is going to be empty. We may need to look at using a MutationObserver.

Reply:
I confirmed that Angular isn't able to find the children.

<img width="1552" alt="Screen Shot 2019-09-05 at 6 20 15 PM" src="https://user-images.githubusercontent.com/330256/64387404-dc405f00-d009-11e9-80db-827471c57288.png">